test/functional/test_short_deck.py

```diff
@@ -172,11 +172,14 @@ def test_flops_are_random():
 
     def _get_flop(state: ShortDeckPokerState) -> List[Card]:
         """Get the public cards for the flop stage."""
-        state = copy.deepcopy(state)
-        while state.betting_stage != "flop":
-            action: Optional[str] = random.choice(state.legal_actions)
-            state = state.apply_action(action)
-        return state._table.community_cards
+        save_state = copy.deepcopy(state)
+        while save_state.betting_stage != "flop":
+            # accounting for when we hit a terminal node before the flop
+            if save_state.betting_stage == "terminal":
```

Comment:
This test was failing because after I changed the end of betting round logic, the action sequences changed for the seed and one possibility is ['fold', 'fold'], which is a terminal node. As such we get an unbound local error, as the action associated with a terminal node is ```None``` - so ```action``` is never defined.